src/Compilers/Core/Portable/Operations/Loops/ForEachLoopOperationInfo.cs

```diff
@@ -19,7 +19,7 @@ internal class ForEachLoopOperationInfo
 
         public readonly bool IsAsynchronous;
         public readonly bool NeedsDispose;
-        public readonly bool KnownToImplementIDisposable;
```

Comment:
>public readonly bool KnownToImplementIDisposable; [](start = 8, length = 49)

I am not comfortable with elimination of this field. There is enough complexity in how it was calculated across both languages that I would prefer us to not take any chances with trying to re-infer it later.